tests/drop_counters/test_configurable_drop_counters.py

```diff
@@ -0,0 +1,214 @@
+        `configurable_drop_counters` package).
+    """
+
+    capabilities = cdc.get_device_capabilities(duthost)
+
+    if not capabilities:
+        pytest.fail("Error fetching device capabilities")
+
+    return capabilities
+
+@pytest.fixture(params=cdc.SUPPORTED_COUNTER_TYPES)
+def setup_counters(request, device_capabilities, duthost):
+    """
+    Returns a method to setup drop counters.
+
+    Notes:
+        This fixture will automatically clean-up created drop counters.
+
+    Returns:
+        A method which, when called, will create a drop counter with the specified drop reasons.
+    """
+
+    if request.param not in device_capabilities["counters"]:
+        pytest.skip("Counter type not supported on target DUT")
+
+    counter_type = request.param
+    supported_reasons = device_capabilities["reasons"][counter_type]
+
+    def _setup_counters(drop_reasons):
+        if any(reason not in supported_reasons for reason in drop_reasons):
+            pytest.skip("Drop reasons not supported on target DUT")
+
+        cdc.create_drop_counter(duthost, "TEST", counter_type, drop_reasons)
+        time.sleep(1)
+
+        return counter_type
+
+    yield _setup_counters
+
+    try:
+        cdc.delete_drop_counter(duthost, "TEST")
+        time.sleep(1)
+    except Exception:
+        logging.info("Drop counter does not exist, skipping delete step...")
+
+@pytest.fixture
+def test_runner(duthost, ptfadapter, testbed_params):
+    """
+    Returns a method to run the drop counter tests.
+
+    Returns:
+        A method which, when called, will send traffic to the DUT and check if the proper
+        drop counter has been incremented.
+    """
+
+    def _runner(counter_type, pkt, rx_port):
+        duthost.command("sonic-clear dropcounters")
+
+        ptf_tx_port_id = testbed_params["phy_intfs"].index(rx_port)
+        _send_packets(duthost, ptfadapter, pkt, ptf_tx_port_id)
+        time.sleep(3)
+
+        recv_count = cdc.get_drop_counts(duthost, counter_type, "TEST", rx_port)
+
+        pytest_assert(recv_count == PACKET_COUNT,
+                      "Expected {} drops, received {}".format(PACKET_COUNT, recv_count))
+
+    return _runner
+
+@pytest.fixture
+def mock_server(ptfhost, fanouthosts, testbed_params):
+    """
+    Mocks the presence of a server beneath a T0.
+
+    Returns:
+        A MockServer which will allow the caller to mock the behavior of a server within
+        a VLAN under a T0.
+    """
+
+    if not testbed_params["vlan_intfs"]:
+        pytest.skip("No VLANs available to mock a server under T0")
+
+    # Randomly pick one of the VLANs beneath the T0.
+    vlan = testbed_params["vlan_intfs"][random.choice(list(testbed_params["vlan_intfs"]))]
+
+    # Generate a dummy IP that falls under the VLAN subnet. Make sure it isn't
+    # the same as the VLAN interface IP!
+    server_ip = vlan["addr"]
+    while server_ip == vlan["addr"]:
+        server_ip = random.choice(list(IPNetwork(vlan["subnet"])))
```

Comment:
Do you care if 2 servers randomized to the same IP?

Reply:
Yes, there is still possibility that the new randomly chosen IP is same as the VLAN interface IP. This kind of random failure would be hard to debug.

Reply:
Fixed!